pkg/engine/masterarmresources.go

```diff
@@ -0,0 +1,154 @@
+		masterResources = append(masterResources, createRouteTable())
+	}
+
+	isPrivateCluster := to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.Enabled)
+
+	if !isPrivateCluster {
+		publicIPAddress := CreatePublicIPAddress()
+		masterResources = append(masterResources, publicIPAddress)
+
+		loadBalancer := CreateLoadBalancer(cs.Properties.MasterProfile.Count)
+		masterResources = append(masterResources, loadBalancer)
+
+		masterNic := CreateNetworkInterfaces(cs)
+		masterResources = append(masterResources, masterNic)
+
+	} else {
+		masterNic := createPrivateClusterNetworkInterface(cs)
+		masterResources = append(masterResources, masterNic)
+
+		provisionJumpBox := cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision()
+
+		if provisionJumpBox {
+			jumpBoxVM := createJumpboxVirtualMachine(cs)
+			masterResources = append(masterResources, jumpBoxVM)
+			jumpboxIsManagedDisks :=
+				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateJumpboxProvision() &&
+					cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.JumpboxProfile.StorageProfile == api.ManagedDisks
+			if !jumpboxIsManagedDisks {
+				jumpBoxStorage := createJumpboxStorageAccount()
+				masterResources = append(masterResources, jumpBoxStorage)
+			}
+			jumpBoxNSG := createJumpboxNSG()
+			masterResources = append(masterResources, jumpBoxNSG)
+			jumpBoxNIC := createJumpboxNIC(cs)
+			masterResources = append(masterResources, jumpBoxNIC)
+		}
+
+	}
+
+	if p.MasterProfile.Count > 1 {
+		internalLb := CreateMasterInternalLoadBalancer(cs)
+		masterResources = append(masterResources, internalLb)
+	}
+
+	isKMSEnabled := to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.EnableEncryptionWithExternalKms)
+
+	if isKMSEnabled {
+		keyVaultStorageAccount := createKeyVaultStorageAccount()
+		masterResources = append(masterResources, keyVaultStorageAccount)
+		keyVault := CreateKeyVault(cs)
+		masterResources = append(masterResources, keyVault)
+	}
+
+	masterVM := CreateVirtualMachine(cs)
+	masterResources = append(masterResources, masterVM)
+
+	useManagedIdentity := cs.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity
+	userAssignedIDEnabled := useManagedIdentity && cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedID != ""
+
+	if useManagedIdentity && !userAssignedIDEnabled {
+		vmasRoleAssignment := createVmasRoleAssignment()
+		masterResources = append(masterResources, vmasRoleAssignment)
+	}
+
+	masterCSE := CreateCustomScriptExtension(cs)
+	masterResources = append(masterResources, masterCSE)
+
+	if cs.IsAksBillingEnabled() {
+		aksBillingExtension := CreateAKSBillingExtension(cs)
+		masterResources = append(masterResources, aksBillingExtension)
+	}
+
+	//TODO: Implement Extensions
+
+	return masterResources
+}
+
+func createKubernetesMasterResourcesVMSS(cs *api.ContainerService) []interface{} {
+	var masterResources []interface{}
+
+	//TODO: Implement CosmosEtcd
```

Comment:
are all the TODOs in this file in-progress work for this PR or TODOs for later after this merged?

Reply:
CosmosDB team needs to publish their APIs :(

Reply:
I am not sure what the answer is to the question.